internal/api/helpers.go

```diff
@@ -232,3 +233,18 @@ func isStringInSlice(checkValue string, list []string) bool {
 func getBodyBytes(req *http.Request) ([]byte, error) {
 	return utilities.GetBodyBytes(req)
 }
+
+func isValidCodeChallenge(codeChallenge string) bool {
+	// See Section 4.2: https://www.rfc-editor.org/rfc/rfc7636#section-4.2
+	hasValidChallengeChars := regexp.MustCompile("^[a-zA-Z-._~0-9]+$").MatchString
+	switch codeChallengeLength := len(codeChallenge); {
+	case codeChallengeLength < 43:
+		return false
+	case codeChallengeLength > 128:
+		return false
+	case !hasValidChallengeChars(codeChallenge):
+		return false
```

Comment:
```suggestion
	case codeChallengeLength < 43, codeChallengeLength > 128, !hasValidChallengeChars(codeChallenge):
		return false
```

Reply:
Wanted to highlight that each case is different (e.g. too long, too short, invalid chars). Am fine either ways though - let's wait for @hf to weigh in

Reply:
I'm fine either way. Whatever you feel like @J0.

Reply:
Let's go with @kangmingtay 's suggestion then